contracts/facets/GmxV2Facet.sol

```
@@ -77,41 +128,26 @@ abstract contract GmxV2Facet is ReentrancyGuardKeccak, OnlyOwnerOrInsolvent {
             })
         );
 
-        IERC20(depositedToken).approve(getGmxV2Router(), tokenAmount);
+        if(longTokenAmount > 0) {
+            IERC20(longToken).approve(getGmxV2Router(), longTokenAmount);
+        }
+        if(shortTokenAmount > 0) {
+            IERC20(shortToken).approve(getGmxV2Router(), shortTokenAmount);
+        }
+
         BasicMulticall(getGmxV2ExchangeRouter()).multicall{ value: msg.value }(data);
 
         // Simulate solvency check
-        {
-            bytes32[] memory dataFeedIds = new bytes32[](2);
-            dataFeedIds[0] = tokenManager.tokenAddressToSymbol(gmToken);
-            dataFeedIds[1] = tokenManager.tokenAddressToSymbol(depositedToken);
-
-            uint256 gmTokenUsdPrice = SolvencyMethods.getPrices(dataFeedIds)[0];
-            uint256 depositTokenUsdPrice = SolvencyMethods.getPrices(dataFeedIds)[1];
-            require(isWithinBounds(
-                depositTokenUsdPrice * tokenAmount / 10**IERC20Metadata(depositedToken).decimals(),  // Deposit Amount In USD
-                minGmAmount * gmTokenUsdPrice / 10**IERC20Metadata(gmToken).decimals())                                                // Output Amount In USD
-            , "Invalid min output value");
-
-            uint256 gmTokensWeightedUsdValue = gmTokenUsdPrice * minGmAmount * tokenManager.debtCoverage(gmToken) / 1e26;
-            require((_getThresholdWeightedValuePayable() + gmTokensWeightedUsdValue) > _getDebtPayable(), "The action may cause the account to become insolvent");
-        }
+        _simulateSolvencyCheck(tokenManager, gmToken, longTokenAmount, shortTokenAmount, minGmAmount, true);
```

Maybe we can do it before the multicall above, to fail fast and save gas>